modules/videoio/src/cap_android_mediandk.cpp

```diff
@@ -257,6 +260,228 @@ class AndroidMediaNdkCapture : public IVideoCapture
     }
 };
 
+
+
+class AndroidMediaNdkVideoWriter CV_FINAL :
+    public cv::IVideoWriter
+{
+    typedef struct {
+        int fourcc;
+        const char* mime;
+        OutputFormat muxerFormat;
+    }
+    FourCCInfo;
+
+    static const int64_t TIMEOUT = 1000L;
+    static const FourCCInfo FOURCC_INFO[];
+
+    static const FourCCInfo* findInfo(int fourcc) {
+        for( const FourCCInfo *it = FOURCC_INFO; NULL != it->mime; it++ ) {
+            if (fourcc == it->fourcc) return it;
+        }
+        return NULL;
+    }
+
+    AMediaFormat* format;
+    AMediaCodec* encoder;
+    AMediaMuxer* muxer;
+    ANativeWindow* surface;
+    long frameIndex;
+    int width;
+    int height;
+    double frameRate;
+    ssize_t videoTrackIndex;
+    int fd;
+
+    void drainEncoder(bool end) {
+        if (end) AMediaCodec_signalEndOfInputStream(encoder);
```

Comment:
> `error: use of undeclared identifier 'AMediaCodec_signalEndOfInputStream'``

Do you know which minimal NDK version is required for this feature?

Reply:
Good point.
I seems to be 26:
https://developer.android.com/ndk/reference/group/media#amediacodec_signalendofinputstream

I'll try to avoid it to make it compile with 21.

Reply:
I added a special implementation for NDK < 26 (actually I think there should be minimum 21).
I think it's slightly slower because there is not conversion RGB => YUV NV12 in opencv.
Also it is limited to "aligned" resolutions (on my device it must be aligned to 4, but it may depends on vendor / hardware).

Reply:
Fixed tailing spaces.